### PR TITLE
Fix bug where offset.copy() != offset

### PR DESCRIPTION
**Pull request by jbrockmendel**
Add test that will currently fail
Make some attributes into cache_readonlys

Last one for now.

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/tseries/test_offsets.py [hunk 1]
```diff
@@ -1952,6 +1952,10 @@ def _check_roundtrip(obj):
         _check_roundtrip(self._object(2))
         _check_roundtrip(self._object() * 2)
 
+    def test_copy(self):
```
Comment (jreback) [inline pandas/tests/tseries/test_offsets.py]: add the issue number (or pr number in this case) as a comment

Comment (jbrockmendel) [reply pandas/tests/tseries/test_offsets.py]: Done.

**Comment by jreback** [comment]
is this orthogonal on #17458 ?

**Comment by jbrockmendel** [comment]
Yes.  They're related in spirit, but need to be fixed separately.

**Comment by jreback** [comment]
thanks; normally I would ask for a whatsnew entry, but I think this is fine.

**Pull request merged**